packages/mdc-checkbox/index.ts

```diff
@@ -21,78 +21,90 @@
  * THE SOFTWARE.
  */
 
-import {StandardJsEventType, getCorrectEventName} from '@material/animation/index.ts';
+import {getCorrectEventName, StandardJsEventType} from '@material/animation/index';
 import MDCComponent from '@material/base/component';
-/* eslint-disable no-unused-vars */
-import {MDCSelectionControlState, MDCSelectionControl} from '@material/selection-control/index';
-/* eslint-enable no-unused-vars */
+import {EventType, SpecificEventListener} from '@material/dom/index';
+import {MDCRipple, MDCRippleFoundation, RippleCapableSurface, util} from '@material/ripple/index';
+import {MDCSelectionControl} from '@material/selection-control/index';
+
 import MDCCheckboxFoundation from './foundation';
-import {MDCRipple, MDCRippleFoundation} from '@material/ripple/index';
-import {getMatchesProperty} from '@material/ripple/util';
 
-/** @const {!Array<string>} */
-const CB_PROTO_PROPS = ['checked', 'indeterminate'];
+const {getMatchesProperty} = util;
+const CB_PROTO_PROPS: string[] = ['checked', 'indeterminate'];
 
 const {ANIMATION_END} = StandardJsEventType;
 
-/**
- * @extends MDCComponent<!MDCCheckboxFoundation>
- * @implements {MDCSelectionControl}
- */
-class MDCCheckbox extends MDCComponent {
-  static attachTo(root) {
+class MDCCheckbox extends MDCComponent<MDCCheckboxFoundation>
+  implements MDCSelectionControl, RippleCapableSurface {
+  static attachTo(root: Element) {
     return new MDCCheckbox(root);
   }
 
   /**
    * Returns the state of the native control element, or null if the native control element is not present.
-   * @return {!MDCSelectionControlState}
-   * @private
    */
-  get nativeCb_() {
+  get nativeCb_(): HTMLInputElement {
     const {NATIVE_CONTROL_SELECTOR} = MDCCheckboxFoundation.strings;
-    const cbEl = /** @type {!MDCSelectionControlState} */ (
-      this.root_.querySelector(NATIVE_CONTROL_SELECTOR));
+    const cbEl = this.root_.querySelector(NATIVE_CONTROL_SELECTOR) as HTMLInputElement;
+    if (!cbEl) {
+      throw new Error(`Checkbox requires a ${NATIVE_CONTROL_SELECTOR} element`);
+    }
     return cbEl;
   }
 
-  constructor(...args) {
-    super(...args);
-
-    /** @private {!MDCRipple} */
-    this.ripple_ = this.initRipple_();
-    /** @private {!Function} */
-    this.handleChange_;
-    /** @private {!Function} */
-    this.handleAnimationEnd_;
-  }
+  root_!: Element;
```

Comment:
This is redundant with `MDCComponent` and missing the `protected` visibility modifier.

Either remove this line or change the type to `protected HTMLElement` and remove the cast below (`(this.root_ as HTMLElement).offsetWidth`).

Reply:
I think we've been opting to keep `root_` as `Element` and cast as necessary so far, right? It looks like we avoid re-declaring this in other components we've converted.

Reply:
Correct. I'm guessing Matt was overriding this as an `HTMLElement` at some point, and then changed it back to `Element` instead of removing it. We should be able to just remove this line.

Reply:
I think this is actually because `root_` is `protected` in the super class, but it must be `public` to fulfill the `RippleCapableSurface` interface. Should definitely add a comment why this is being done though

Reply:
I was wrong: public visibility for this property is required by `RippleCapableSurface`.

I'll add a comment to clarify this.